internal/dbtest/db_test.go

```diff
@@ -2000,3 +2005,40 @@ func testWithPointerPrimaryKeyHasManyWithDriverValuer(t *testing.T, db *bun.DB)
 	require.NoError(t, err)
 	require.Len(t, program.Items, 2)
 }
+
+func testRelationJoinDataRace(t *testing.T, db *bun.DB) {
+	if db.Dialect().Name().String() != pgName {
+		return
```

Comment:
Is the test PG-specific, or do we skip other dialects because we only need to test with one?

If it's the latter, would it be clearer to just make this a separate test altogether?
Like [`TestMssqlMerge`](https://github.com/uptrace/bun/blob/7fc6639dea51f98c1f87c24a82609844edc74252/internal/dbtest/mssql_test.go#L9), which explicitly uses only 1 dialect.

Reply:
Similar to the previous issue, this test isn't theoretically specific to PostgreSQL. However, these tests currently exhibit unrelated failures under the race detector, making it difficult to isolate the intended behavior. Perhaps I should add a TODO note here to indicate that these tests need to be revisited and fixed for other dialects in the context of race condition detection?

Reply:
Agree, a comment next to that if-block explaining why we only test for a single dialect now and how we should address that in the future would be good!